### PR TITLE
Bugfix/hide user registered events homescreen

### PR DESCRIPTION
**Pull request by winniezheng**
https://trello.com/c/elZyUsAW/173-bug-when-the-user-registers-for-an-event-in-the-home-screen-it-should-not-show-up-in-the-list-of-results

### CODE REVIEW

File: src/screens/Home/EventList/EventList.tsx [hunk 1]
```diff
@@ -74,7 +107,7 @@ const EventList: React.FC<EventListProps> = ({
   return (
     <FlatList
       data={
-        data?.getFoodDrives.filter((foodDrive) =>
+        foodDrives?.filter((foodDrive) =>
           foodDrive.name.toLowerCase().includes(search.toLowerCase()),
```
Comment (chrisrabe) [inline src/screens/Home/EventList/EventList.tsx:111]: I think we can also add the search filter logic into the useMemo

File: src/screens/Home/EventList/EventList.tsx [hunk 2]
```diff
@@ -23,22 +25,53 @@ interface EventListProps extends EventListPropsFromRedux {
...
   const toastProvider = useToastProvider();
   const styles = useStyles();
 
+  const currentUserRegisteredEvents = invitationData?.getInvitations.map(
+    (invitation) => invitation.event.id,
+  );
+  //get food drives excluding current user's registered events
+  const foodDrives = useMemo(
+    () =>
+      data?.getFoodDrives.filter((foodDrive) => {
```
Comment (chrisrabe) [inline src/screens/Home/EventList/EventList.tsx:56]: This approach works, but kind of defeats the purpose of GraphQL's ability to support FE queries declaratively. For this ticket, we could update the BE to take in a variable that specifies that we want to exclude events that the user is invited to and then modify the `GET_FOOD_DRIVES` query to take in that variable.

Also, having two live queries might affect performance of the app.

Comment (chrisrabe) [reply src/screens/Home/EventList/EventList.tsx:56]: I'll put this in another ticket

**Review by chrisrabe — CHANGES_REQUESTED** [review]
Search filter could also be moved to the useMemo hook

File: src/screens/Home/EventList/EventList.tsx (hunk 2 above)
Comment (winniezheng) [reply src/screens/Home/EventList/EventList.tsx:56]: I'll cancel this PR and do that ^

**Pull request closed without merging**